com.github.christianrauch.Jahresarbeit-2003.yaml

```diff
@@ -0,0 +1,121 @@
+app-id: com.github.christianrauch.Jahresarbeit-2003
+runtime: org.freedesktop.Platform
+runtime-version: "21.08"
+sdk: org.freedesktop.Sdk
+command: /app/ja2003
+finish-args:
+- --socket=wayland
+- --socket=fallback-x11
+- --device=dri
+cleanup:
+- /include
+- /bin
+- /lib/cmake
+- /lib/debug
+- /lib/pkg-config
+- /share/aclocal
+- "*.a"
+
+modules:
+  - name: libdecor
+    sources:
+    - type: git
+      url: https://gitlab.gnome.org/jadahl/libdecor.git
+      tag: "0.1.0"
+    buildsystem: meson
+    builddir: true
+
+  - name: SDL2
+    sources:
+    - type: archive
+      url: https://github.com/libsdl-org/SDL/archive/refs/tags/release-2.0.18.tar.gz
+      sha256: b7061be3d86d14cdddbc696ab5b377ed28418f33be256c92215a57363329aa5f
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+    - -DCMAKE_BUILD_TYPE=Release
+    # subsystems
+    - -DSDL_ATOMIC=OFF
+    - -DSDL_AUDIO=OFF
+    - -DSDL_VIDEO=ON
+    - -DSDL_RENDER=ON
+    - -DSDL_EVENTS=ON
+    - -DSDL_JOYSTICK=ON
+    - -DSDL_HAPTIC=ON
+    - -DSDL_HIDAPI=OFF
+    - -DSDL_POWER=OFF
+    - -DSDL_THREADS=OFF
+    - -DSDL_TIMERS=OFF
+    - -DSDL_FILE=OFF
+    - -DSDL_LOADSO=ON
+    - -DSDL_CPUINFO=OFF
+    - -DSDL_FILESYSTEM=ON
+    - -DSDL_DLOPEN=OFF
+    - -DSDL_SENSOR=OFF
+    - -DSDL_LOCALE=OFF
+    # Wayland build with libdecor support
+    - -DSDL_WAYLAND_LIBDECOR=ON
+
+  - name: SDL2-ttf
+    sources:
+    - type: archive
+      url: https://github.com/libsdl-org/SDL_ttf/archive/refs/tags/release-2.0.15.tar.gz
+      sha256: 02e887b560faf398cbd60f56ce0a1cbaf96012dd4ddaa455f8307cb4911c86cc
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+    - -DCMAKE_BUILD_TYPE=Release
+
+  - name: SDL2-image
+    sources:
+    - type: git
+      url: https://github.com/libsdl-org/SDL_image.git
+      commit: a31da6b
+    buildsystem: autotools
+    config-opts:
+    - --disable-bmp
+    - --enable-gif
+    - --enable-jpg
+    - --disable-jpg-shared
+    - --disable-lbm
+    - --disable-pcx
+    - --disable-png
+    - --disable-png-shared
+    - --disable-pnm
+    - --disable-svg
+    - --disable-tga
+    - --disable-tif
+    - --disable-tif-shared
+    - --disable-xcf
+    - --disable-xpm
+    - --disable-xv
+    - --disable-webp
+    - --disable-webp-shared
+
+  - name: JA2003
+    sources:
+    - type: git
+      url: https://github.com/christianrauch/Jahresarbeit-2003.git
+      tag: "2.0"
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+    - -DCMAKE_BUILD_TYPE=Release
+
+  - name: launcher
```

Comment:
usually the metadata file are declared / installed with the main module.

Reply:
Most of the metadata (AppData, *.desktop files) encode the flatpak specific ID. So I am maintaining them in the dedicated flatpak repo outside of the generic main repo.

Reply:
1, this is irrelevant to what I just said here.
2. lot of package don't have to touch the appdata or desktop file and just use upstream, sometime with just a renae. Since you are upstream maybe a future release can work using upstream metadata.